### PR TITLE
Meta: Add a minimal SDL2 based videoplayer for Lagom

### PR DESCRIPTION
**Pull request by feliwir**
Adds a simple SDL2 application for testing `LibVideo` on the host system / via lagom.
I chose `Meta/Lagom/Contrib` as the directory, since `MacPDF` sits there aswell

### CODE REVIEW

File: Meta/Lagom/Contrib/VideoPlayerSDL/main.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        warnln("Failed to decode file {}", filename);
+        return 1;
+    }
+
+    SDL_Texture* texture = NULL;
+
+    auto playback_manager = load_file_result.release_value();
+    playback_manager->on_video_frame = [&texture](RefPtr<Gfx::Bitmap> frame) {
+        u8* pixels = frame->scanline_u8(0);
+        int result = SDL_UpdateTexture(texture, NULL, pixels, frame->pitch());
```
Comment (Zaggy1024) [inline Meta/Lagom/Contrib/VideoPlayerSDL/main.cpp:40]: If `SDL_UpdateTexture` expects a buffer sized to `w * h * 4` bytes (if that's the size of the pixel format), then this will overread if the current frame doesn't match the initial video resolution.

Note that the `Track::video_data()` is not really accurate, the values there can change at any time, including on the first decoded frame.

To avoid this, I would create/recreate the texture inside `on_video_frame` every time the video resolution changes. I suppose that if the texture is copied 1:1 to the window, you'll have to resize the window at the same time.

**Review by Zaggy1024 — COMMENTED** [review]
Overall looks good to me, except one concern. If the CMake stuff looks good, this should be fine to merge.

**Comment by Zaggy1024** [comment]
LGTM!

**Comment by ADKaster** [comment]
We will also want to add CI testing for this application, as none of our bots have SDL2 installed.
